### PR TITLE
Add course ans status scheme. Add rest of APIs

### PR DESCRIPTION
**Pull request by cookie777**

### CODE REVIEW

File: docs/sample.yml [hunk 1]
```diff
@@ -473,118 +573,175 @@ components:
...
+            The date when the course ends.
         class_schedules:
           type: array
           items:
             properties:
               start_time:
                 type: string
-                format: dd/mm/yyyy
+                format: yyyy/MM/dd HH:mm:ss.sss
+                example: "2021-09-01 16:00.000"
```
Comment (DaiSugi01) [inline docs/sample.yml:645]: It should be `2021-09-01 16:00:00.000`

File: docs/sample.yml [hunk 2]
```diff
@@ -473,118 +573,175 @@ components:
...
               end_time:
                 type: string
-                format: dd/mm/yyyy
+                format: yyyy/MM/dd HH:mm:ss.sss
+                example: "2021-09-01 18:00.000"
```
Comment (DaiSugi01) [inline docs/sample.yml:649]: It should be 2021-09-01 18:00:00.000

File: docs/sample.yml [hunk 3]
```diff
@@ -473,118 +573,175 @@ components:
...
+        created_date:
+          type: string
+          format: yyyy/MM/dd HH:mm:ss.sss
+          example: "2021-06-17 20:46:11.716"
+          description: |
+            Timestamp which this data was first created. Following to ISO 8601 format. yyyy/MM/dd hh:mm:ss
         updated_date:
           type: string
-          format: yyyy-MM-dd HH:mm:ss.sss
+          format: yyyy/MM/dd HH:mm:ss.sss
```
Comment (DaiSugi01) [inline docs/sample.yml]: The format should be `yyyy-MM-dd HH:mm:ss.sss`

File: docs/sample.yml [hunk 4]
```diff
@@ -473,118 +573,175 @@ components:
...
-            format: dd/mm/yyyy
           description:
-            A list of available course schedules
+            A list of available course schedules.
+        created_date:
+          type: string
+          format: yyyy/MM/dd HH:mm:ss.sss
+          example: "2021-06-17 20:46:11.716"
+          description: |
+            Timestamp which this data was first created. Following to ISO 8601 format. yyyy/MM/dd hh:mm:ss
```
Comment (DaiSugi01) [inline docs/sample.yml]: The format should be yyyy-MM-dd HH:mm:ss.sss

File: docs/sample.yml [hunk 5]
```diff
@@ -473,118 +573,175 @@ components:
...
         updated_date:
           type: string
-          format: yyyy-MM-dd HH:mm:ss.sss
+          format: yyyy/MM/dd HH:mm:ss.sss
+          example: "2021-06-18 15:22:05.292"
           description: |
-            updated date
-        created_date:
+            Timestamp which this data was last updated. Following to ISO 8601 format. yyyy/MM/dd hh:mm:ss
```
Comment (DaiSugi01) [inline docs/sample.yml]: The format should be `yyyy-MM-dd HH:mm:ss.sss`

File: docs/sample.yml [hunk 6]
```diff
@@ -473,118 +573,175 @@ components:
...
+            - archived
+              - A course which has already finished
+            - active
+              - A course which is now on going
+            - inactive
+              - A course which is will be "active" in the future. If an user apply for the course which has this status, it will be "wait list"
         start_date:
           type: string
-          format: dd/mm/yyyy
+          format: yyyy/MM/dd HH:mm:ss.sss
```
Comment (DaiSugi01) [inline docs/sample.yml]: The format should be `yyyy-MM-dd HH:mm:ss.sss`

File: docs/sample.yml [hunk 7]
```diff
@@ -473,118 +573,175 @@ components:
...
+          example: "2021-09-01 00:00:00.000"
           description: |
-            course start date
+            The date when the course starts.
         end_date:
           type: string
-          format: dd/mm/yyyy
+          format: yyyy/MM/dd HH:mm:ss.sss
```
Comment (DaiSugi01) [inline docs/sample.yml]: The format should be `yyyy-MM-dd HH:mm:ss.sss`

File: docs/sample.yml [hunk 8]
```diff
@@ -473,118 +573,175 @@ components:
...
+          example: [ "Tuesday 7pm - 8:30pm", "Thursday 7pm - 8:30pm", "Saturday 1pm - 4pm"]
           items:
             type: string
-            format: dd/mm/yyyy
           description:
-            A list of available course schedules
+            A list of available course schedules.
+        created_date:
+          type: string
+          format: yyyy/MM/dd HH:mm:ss.sss
```
Comment (DaiSugi01) [inline docs/sample.yml]: The format should be `yyyy-MM-dd HH:mm:ss.sss`

File: docs/sample.yml [hunk 9]
```diff
@@ -376,6 +445,37 @@ paths:
...
               schema:
                 $ref: "#/components/schemas/Error"
+  
+  #Tak
+  /courses/{id}/user:
+    get:
+      tags:
+        - courses
+      description: Get all user-ids that a course has.
+      responses:
```
Comment (DaiSugi01) [inline docs/sample.yml:462]: need request parameters under the description like below example

      parameters:
      - name: id
        in: path
        description: Course ID
        required: true
        schema:
          type: string

File: docs/sample.yml [hunk 10]
```diff
@@ -376,6 +445,37 @@ paths:
...
+        200:
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Status"
+                type: array  
+              example:
+                - course_id: 442c16a8a353
```
Comment (DaiSugi01) [inline docs/sample.yml]: how about using `DPCS100#1, DPCS101#1` for course_id

File: docs/sample.yml [hunk 11]
```diff
@@ -209,6 +209,89 @@ paths:
               schema:
                 $ref: "#/components/schemas/Error"
 
+  # Tak
+  /user/{token}/course:
```
Comment (DaiSugi01) [inline docs/sample.yml]: need to talk about the token id as a request parameter.

File: docs/sample.yml [hunk 12]
```diff
@@ -209,6 +209,89 @@ paths:
...
+
+        default:
+          description: error occurred
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+
+  # Tak
+  /user/{token}/course/{id}:
```
Comment (DaiSugi01) [inline docs/sample.yml]: need to talk about the token id as a request parameter.

**Review by DaiSugi01 — CHANGES_REQUESTED** [review]

File: docs/sample.yml (hunk 1 above)
Comment (cookie777) [reply docs/sample.yml:645]: My mistake!

File: docs/sample.yml (hunk 10 above)
Comment (cookie777) [reply docs/sample.yml]: it's user_id.
Typo!

**Comment by cookie777** [comment]
Change
- amend all yyyy/MM/dd -> yyyy-MM-dd
- course_id -> user_id in line 471
- append parameters in 455
- comment out {token} api

**Review by DaiSugi01 — APPROVED** [review]

**Pull request merged**